mu4/notation/view/notationviewinputcontroller.cpp

```diff
@@ -85,6 +100,30 @@ void NotationViewInputController::setZoom(int zoomPercentage, const QPoint& pos)
     m_view->setZoom(zoomPercentage, pos);
 }
 
+void NotationViewInputController::setViewMode(const ViewMode& viewMode)
+{
+    auto notation = currentNotation();
+    if (!notation) {
+        return;
+    }
+
+    notation->setViewMode(viewMode);
+}
+
+void NotationViewInputController::toggleViewMode()
+{
+    auto notation = currentNotation();
+    if (!notation) {
+        return;
+    }
+
+    if (notation->getViewMode() != ViewMode::PAGE) {
```

Comment:
Why are we not considering "ViewMode::System here"? I'm a bit worried about this "toggleMethod"
Probably need some explanation

Reply:
I mentioned this in our Telegram chat but I'll leave it here too: toggleViewMode handles keyboard shortcuts (ctrl+shift+V, which activates a "viewmode" action), while setViewMode(ViewMode) handles actions sent when the user clicks the ComboBox. I implemented the keyboard shortcut this way to match how MU3 handles the same keyboard shortcut (just toggling between Line and Page, skipping System), but I can either remove keyboard shortcut handling or have it cycle through all three options.

Reply:
I believe that what vasily meant by "probably needs some explanation" was that it would be great to have your last comment as a code comment.

Reply:
Gotcha, thanks for the clarification! I ended up removing the method altogether at Vasily's request since there are plans to re-work the keyboard shortcuts anyway. That should clear up the confusion!